File: ios/ReactNativeWallpaperManager.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(ReactNativeWallpaperManager, NSObject)

RCT_EXTERN_METHOD(multiply:(float)a withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(applyWallpaper:(NSString *)uri 
          screen:(NSString *)screen 
          withResolver:(RCTPromiseResolveBlock)resolve 
          withRejecter:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}


@end

File: ios/ReactNativeWallpaperManager.swift
@objc(ReactNativeWallpaperManager)
class ReactNativeWallpaperManager: NSObject {

  @objc(multiply:withB:withResolver:withRejecter:)
  func multiply(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
    resolve(a*b)
  }

  @objc(applyWallpaper:screen:withResolver:withRejecter:)
  func applyWallpaper(uri: String, 
      screen: String, 
      resolve: RCTPromiseResolveBlock,
      reject: RCTPromiseRejectBlock) -> Void {
        resolve("")
  }
}
